packages/nova-react/src/graphql/hooks.ts

```diff
@@ -324,22 +326,27 @@ export function useSubscription<TSubscriptionPayload extends OperationType>(
   graphql.useSubscription(config);
 }
 
-interface MutationCommitterOptions<TMutationPayload extends OperationType> {
+interface MutationCommitterOptionsBase<TMutationPayload extends OperationType> {
   variables: TMutationPayload["variables"];
   /**
    * Should be avoided when possible as it will not be compatible with Relay APIs.
    */
   context?: TMutationPayload["context"];
   optimisticResponse?: Partial<TMutationPayload["response"]> | null;
+  onError?: (error: Error) => void;
+}
+
+interface MutationCommitterOptions<TMutationPayload extends OperationType> extends MutationCommitterOptionsBase<TMutationPayload> {
+  onCompleted?: ((response: TMutationPayload["response"], errors: PayloadError[] | null) => void | null) | undefined;
+}
+
+interface MutationCommitterOptions_deprecated<TMutationPayload extends OperationType> extends MutationCommitterOptionsBase<TMutationPayload> {
   onCompleted?: (response: TMutationPayload["response"]) => void;
 }
```

Comment:
I'm not sure on the inheritance vs just duplicating the type for the options. I'd probably say we should have two? 

My thinking is that when adding more Relay specific options you'd need to make sure to add them to the Options one and not the Base (as that would also change the deprecated one. The other thing is when removing the deprecated one, we could also just remove the type for it. Then we would be left with just the "new" one, instead of the new one and the base (which we'd then probably merge back?)

That said, I'm ok with it if you like to keep it like this.

Reply:
Sure, I can keep two of them